sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receipt/_claims_digest_computation.py

```diff
@@ -0,0 +1,192 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Algorithm implementation for computing Azure Confidential Ledger application claims."""
+
+from base64 import b64decode
+from hashlib import sha256
+import hmac
+from typing import Dict, List, Any
+
+from azure.confidentialledger.receipt._claims_models import (
+    ApplicationClaim,
+    LedgerEntryClaim,
+    ClaimDigest,
+)
+
+LEDGER_ENTRY_CLAIM_TYPE = "LedgerEntry"
+DIGEST_CLAIM_TYPE = "ClaimDigest"
+LEDGER_ENTRY_V1_CLAIM_PROTOCOL = "LedgerEntryV1"
+
+
+def compute_claims_digest(application_claims: List[Dict[str, Any]]) -> str:
+    """
+    Compute the claims digest from a list of Azure Confidential Ledger application claims.
+
+    :param application_claims: List of application claims to be verified against the receipt.
+    :type application_claims: List[Dict[str, Any]]
+
+    :return: The claims digest of the application claims.
+    :rtype: str
+    :raises ValueError: If the claims digest computation has failed.
+    """
+
+    # Validate application claims provided by the user
+    _validate_application_claims(application_claims)
+
+    # Convert application claims JSON objects to ApplicationClaim model
+    application_claims_obj = []
+    for claim_dict in application_claims:
+        claim = ApplicationClaim.from_dict(claim_dict)
+        application_claims_obj.append(claim)
+
+    # Compute claims digest from application claims
+    return _compute_claims_hexdigest(application_claims_obj)
+
+
+def _validate_application_claims(application_claims: List[Dict[str, Any]]):
+    """Validate the application claims in a write transaction receipt."""
+
+    assert isinstance(application_claims, list)
+    assert len(application_claims) > 0, "Application claims list cannot be empty"
```

Comment:
Is this necessary? Does the logic not handle empty claims (e.g. for transactions before we started attaching claims)?

Reply:
Never mind, I see - we only verify claims if they are present, so this is a good thing to check here